orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -887,15 +887,8 @@ common::Status SplitGraph(Graph& graph,
       send_input_args.push_back(updated_node_arg);
 
       auto dtype = original_node_arg->TypeAsProto()->tensor_type().elem_type();
-      switch (dtype) {
-        case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
-          element_types.add_ints(static_cast<int64_t>(1));
-          break;
-        default:
-          // Assume all tensors are of type float.
-          // TODO: update if graph supports other data type.
-          ORT_THROW("pipeline partition unsupported 'type' value: ", dtype);
-      }
+
+      element_types.add_ints(static_cast<int64_t>(dtype));
```

Comment:
>element_types.add_ints(static_cast<int64_t>(dtype)); [](start = 6, length = 52)

What happens to strings?

Reply:
What do you mean for strings? Any elem_type that falls into declared TensorProto DataTypes, should be able to be added to element_types. 

---
In reply to: [436922534](https://github.com/microsoft/onnxruntime/pull/4113#discussion_r436922534) [](ancestors = 436922534)